process-engine/process_engine_core
# ✨ Retry send task

## Changes

1. Parse `retryIntervalInMs` and `maxRetries` extension properties for SendTasks
2. Implement Retry mechanism for SendTasks
    - The handler will now keep sending the message, until either the maximum number of retries is exceeded or a response was received
    - Retry interval defaults to 500ms
    - Max number of retries defaults to -1 - i.e. `unlimited`

## Issues

Part of https://github.com/process-engine/process_engine_runtime/issues/538

PR: #325

## How to test the changes

- Start a ProcessInstance with a SendTask
- See that the SendTask periodically sends its message
- Start a ProcessInstance with a matching ReceiveTask
- See that the SendTask will continue, after the ReceiveTask acknowledges receiving the message
